aws/adapter.go

```diff
@@ -636,39 +636,22 @@ func (a *Adapter) UpdateAutoScalingGroupsAndInstances() error {
 	}
 	a.singleInstances = newSingleInstances
 
-	// update ASGs (create new map to get rid of deleted ASGs)
-	newAutoScalingGroups := make(map[string]*autoScalingGroupDetails)
-	autoScalingGroupsToFetchMap := make(map[string]bool)
 	for instanceID, details := range a.ec2Details {
-		asgName, err := getAutoScalingGroupName(details.tags)
+		_, err := getAutoScalingGroupName(details.tags)
 		if err != nil {
 			// Instance is not in ASG, save in single instances list.
 			a.singleInstances[instanceID] = details
 			continue
 		}
-		if _, ok := newAutoScalingGroups[asgName]; !ok {
-			if _, ok := a.autoScalingGroups[asgName]; ok {
-				newAutoScalingGroups[asgName] = a.autoScalingGroups[asgName]
-			} else {
-				// Save ASGs that have to be loaded to load all of them in one API call
-				autoScalingGroupsToFetchMap[asgName] = true
-			}
-		}
-	}
-
-	autoScalingGroupsToFetch := make([]string, 0, len(autoScalingGroupsToFetchMap))
-	for asgName := range autoScalingGroupsToFetchMap {
-		autoScalingGroupsToFetch = append(autoScalingGroupsToFetch, asgName)
 	}
 
-	if len(autoScalingGroupsToFetch) != 0 {
-		fetchedAutoScalingGroups, err := getAutoScalingGroupsByName(a.autoscaling, autoScalingGroupsToFetch)
-		if err != nil {
-			log.Errorf("failed fetching Auto Scaling Groups details: %v", err)
-		} else {
-			for name, asg := range fetchedAutoScalingGroups {
-				newAutoScalingGroups[name] = asg
-			}
+	newAutoScalingGroups := make(map[string]*autoScalingGroupDetails)
+	fetchedAutoScalingGroups, err := getOwnedAutoScalingGroups(a.autoscaling)
+	if err != nil {
+		log.Errorf("failed fetching Auto Scaling Groups details: %v", err)
```

Comment:
We should return this as an error instead of logging it here to be safe. If we just log it then we will set `a.autoScalingGroups` to an empty map below which would be wrong. Before logging kind a worked because it would just set the map to the same value as it already was in case of error on this call.

Reply:
yes, you're right. changing..